test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SpecializationE2ETests.cs

```diff
@@ -246,6 +243,51 @@ public async Task Specialization_ResetsSharedLoadContext()
             }
         }
 
+        [Fact]
+        public async Task Specialization_ResetsSecretManagerRepository()
```

Comment:
Are there any edge cases we can test for as well?

Reply:
Trying to think through some but can't come up with any that seem worthwhile. We're relying on the `Options` infrastructure to handle this (now that it's properly wired up) and the usage of `Lazy` so I think most of the corner cases are handled by those things.

Any ideas you had in mind?

Reply:
nope :D